Sources/MediaScrubProgressBar.swift

```diff
@@ -0,0 +1,167 @@
+    }()
+    
+    private lazy var remainingTimeLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont.systemFont(ofSize: 12)
+        label.textColor = .white
+        label.text = "--:--"
+        label.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        return label
+    }()
+
+    private lazy var scrubbingIndicatorLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont.systemFont(ofSize: 12)
+        label.textColor = .white
+        label.textAlignment = .center
+        label.setContentHuggingPriority(.defaultLow, for: .vertical)
+        return label
+    }()
+
+    private lazy var scrubbingHelpLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont.systemFont(ofSize: 10)
+        label.textColor = .white
+        label.text = NSLocalizedString("PLAYBACK_SCRUB_HELP", comment: "")
+        label.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        label.textAlignment = .center
+        return label
+    }()
+
+    private lazy var scrubLabelStackView: UIStackView = {
+        let scrubLabelStack = UIStackView(arrangedSubviews: [scrubbingIndicatorLabel, scrubbingHelpLabel])
+        scrubLabelStack.axis = .vertical
+        scrubLabelStack.isHidden = true
+        scrubLabelStack.backgroundColor = UIColor(0x000000, 0.6)
+        return scrubLabelStack
+    }()
+    
+    // MARK: Initializers
+    required init(coder: NSCoder) {
+        fatalError("init(coder: NSCoder) not implemented")
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupViews()
+    }
+    
+    // MARK: Instance Methods
+    private func setupViews() {
+        let horizontalStack = UIStackView(arrangedSubviews: [elapsedTimeLabel, remainingTimeLabel])
+        horizontalStack.distribution = .equalSpacing
+        addArrangedSubview(scrubLabelStackView)
+        addArrangedSubview(horizontalStack)
+        addArrangedSubview(progressSlider)
+        spacing = 5
+        axis = .vertical
+        translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    private func updateScrubLabel() {
+        switch progressSlider.scrubbingSpeed {
+        case 1.0:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_HIGH", comment:"")
+        case 0.5:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_HALF", comment: "")
+        case 0.25:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_QUARTER", comment: "")
+        default:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_FINE", comment: "")
+        }
+    }
+
+    @objc private func updatePlaybackPosition() {
+        if !positionSet {
+            playbackController.playbackPosition = progressSlider.value
+            playbackController.setNeedsMetadataUpdate()
+            positionSet = true
+        }
+    }
+
+    @objc func updateForScrubbing() {
```

Comment:
can we just name it updateUI because updateforscrubbing sounds like you update in preparation for scrubbing.